### DISCUSSION TITLE
How to display time on timescale (intraday data)?

### DISCUSSION THREAD

**Discussion by timocov** [tradingview/lightweight-charts#642] [Q&A] [answered]
I have intraday data (within a day), is it possible to display time on the time scale?

![image](https://user-images.githubusercontent.com/3112183/101876543-96276300-3b9d-11eb-879e-d2bc10a15813.png)

**Comment by timocov** [comment, marked as answer]
Yes, it's possible.

Firstly, you need to be sure, that you provide proper `time` value for every data item - it should be [unix timestamp (number)](https://github.com/tradingview/lightweight-charts/blob/v3.7.0/docs/time.md#unix-timestamp).

Secondly, you need to set up time scale to show time part of the dates. It could be done via providing [a proper options](https://github.com/tradingview/lightweight-charts/blob/v3.7.0/docs/time-scale.md#time-scale-options):
- set `timeVisible` to `true` if you'd like to display time (hours and minutes) on time scale
- set `secondsVisible` to `true` if you'd like to display seconds along with hours and minutes (note that it won't affect anything if `timeVisible` is disabled)

See intraday data example https://jsfiddle.net/TradingView/bmL0vont/.

**Reply by berkayyildiz** [reply]
I want to add, don't forget that JS Date.getTime() method returns a timestamp that includes milliseconds but unix timestamp not includes that, when you use JS timestamp chart displays data oddly. You can do something like this to convert JS timestamp to unix timestamp `Math.floor(new Date().getTime() / 1000)`.

**Reply by cegv14** [reply]
Muchas gracias tu ejemplo me sirvio mucho, THANKSS

**Reply by jrm0695** [reply]
Any possibility to display also millisecond in the time axis ?

**Reply by timocov** [reply]
> Any possibility to display also millisecond in the time axis ?

@jrm0695 see https://tradingview.github.io/lightweight-charts/docs/api/interfaces/TimeScaleOptions#tickmarkformatter and https://tradingview.github.io/lightweight-charts/docs/api/interfaces/LocalizationOptions#timeformatter

**Reply by tiger2014** [reply]
@timocov Thanks for your [https://jsfiddle.net/TradingView/bmL0vont/]. And I want to know if we can add our backtest buy/sell signal to the chart, and if we can add our data to the chart like the red box in the screenshot:

![image](https://user-images.githubusercontent.com/6703612/185814062-246b9714-1f8a-4b50-8cc4-f9e05c7e0efa.png)

Thanks!

**Reply by osmar-pj** [reply]
Thank u!

**Reply by Kenildev007** [reply]
thanks bro it help me solve the problem

**Comment by hafidzalafaf** [comment]
thanks bro!
